packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,396 @@
+import { render, screen } from '@testing-library/angular';
+import '@testing-library/jest-dom';
+import { clearElements } from '../utils';
+import { UtrechtCheckboxAttr } from './component';
+
+afterEach(() => {
+  clearElements();
+});
+
+describe('Checkbox', () => {
+  it('renders a checkbox role element', async () => {
+    await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    const checkbox = screen.getByRole('checkbox');
+
+    expect(checkbox).toBeInTheDocument();
+    expect(checkbox).toBeVisible();
+  });
+
+  it('renders an HTML input type=checkbox element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    expect(fixture.nativeElement).toBeInTheDocument();
+  });
+
+  it('displays as CSS block element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    expect(fixture.nativeElement).toBeVisible();
+    expect(fixture.nativeElement).toHaveStyle({ display: 'block' });
+  });
+
+  it('renders a design system BEM block class name', async () => {
+    const { container } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    const checkbox = container.querySelector(':only-child');
+    expect(checkbox).toHaveClass('utrecht-checkbox');
+  });
+
+  describe('checked variant', () => {
+    it('is not checked by default', async () => {
+      await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).not.toBeChecked();
+    });
+
+    it('omits non-essential checked attributes when not checked', async () => {
+      await render('<input checked="false" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toHaveAttribute('aria-checked');
+      expect(checkbox).toHaveAttribute('checked');
+    });
+
+    it('can have a checked state', async () => {
+      const handleChange = jest.fn();
+      await render('<input (change)="sendValue($event)" [checked]="checked" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          sendValue: handleChange,
+          checked: true,
+        },
+      });
```

Comment:
I think the approach for a checked state is great, but the `(change)` event handler should be moved to a completely separate test. Has nothing to do with the checked state per se.